<?xml version="1.0" encoding="UTF-8"?>
<launch>
<!-- TODO:通过yaml传参 -->
  <!-- <rosparam file="$(find lidar_align)/config/params.yaml" command="load" /> -->
  <arg name="bag_file" default="/home/zh/dataset/jz/iplus_bag_S9_filter.bag"/>   <!-- 需要标定的目标bag绝对路径 -->
  
  <arg name="transforms_from_csv" default="false"/>
  <arg name="csv_file" default="PATH/TO/YOUR.csv"/>

  <node pkg="lidar_align" name="lidar_align" type="lidar_align_node" output="screen">
    <param name="input_bag_path" value="$(arg bag_file)" /> 
    <param name="input_csv_path" value="$(arg csv_file)" />
    <param name="initial_pose" value="0.0 0.0 0.0 0.0 0.0 0.0 0.0" />     <!-- 7个外参分别为x y z roll pitch yawl 时间偏移(默认为0)  -->
    <param name="use_LidarScan_number" value="50" />       <!-- 使用的雷达帧数 -->
    <param name="max_evals" value="300" />     <!-- 最大迭代次数 -->
    <param name="output_pointcloud_path" value="$(find lidar_align)/results/lidar.ply" />   <!-- 映射地图的输出路径 -->
    <param name="output_calibration_path" value="$(find lidar_align)/results/$(anon calibration).txt" />    <!-- 标定结果的输出路径 -->
    <param name="transforms_from_csv" value="$(arg transforms_from_csv)"/>
  </node>

</launch>